OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -96,7 +96,11 @@ public void Attack(Actor targetActor, bool allowMove = true, bool forceAttack =
 			if (!targetActor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
-			attackBase.AttackTarget(target, true, allowMove, forceAttack);
+			var attack = attackBase.FirstOrDefault(x => !x.IsTraitDisabled && x.HasAnyValidWeapons(target));
+			if (attack == null)
+				throw new LuaException("Actor {0} was ordered to attack {1} but has no active Attack* trait to do so!.".F(Self, targetActor));
```

Comment:
I don't know, it probably could be done in some for loop that attacks units produced by Lua code or something in which case we want it to silently fail or have some way to ignore the error.

It depends on what we want the method to do so it probably needs some discussion on IRC or you get pchote to chip in.